### PR TITLE
replace broken coveralls by codecov

### LINKED ISSUE #93: coveralls: fix status
Travis is uploading builds to coveralls, but the coveralls UI isn't updating with the most recent builds. In fact it only shows the first build.
- travis job (Nov 27 2018): https://travis-ci.org/tpm2-software/tpm2-pkcs11/jobs/460507260
- coveralls: https://coveralls.io/github/tpm2-software/tpm2-pkcs11?branch=master

### PR DESCRIPTION
**Pull request by joholl**
Fixes #93 by replacing the broken coveralls coverage with codecov. The repository needs to be linked with codecov and the token needs to be updated in .travis.yml to make it work.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,9 +6,17 @@ compiler:
 - gcc
 - clang
 env:
-- PYTHON_INTERPRETER=python2
-- PYTHON_INTERPRETER=python3
+  matrix:
+    - PYTHON_INTERPRETER=python2
+    - PYTHON_INTERPRETER=python3
+  global:
+    - CODECOV_TOKEN=xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
```
Comment (joholl) [inline .travis.yml]: This token needs to be updated

**Comment by williamcroberts** [comment]
LGTM ill cherry pick and push this back. I'm going to set CODECOV_TOKEN in Travis.

**Comment by PeterHuewe** [comment]
Ping

**Comment by williamcroberts** [comment]
I haven't forgotten about this, I was waiting for permissions for the application to our org, which I think is done.

**Comment by williamcroberts** [comment]
I have this staged on the codecov branch https://github.com/tpm2-software/tpm2-pkcs11/tree/codecov

Any further work can be done on the issue tracker or as additional PR's. I need to do it on a trusted branch so the CODECOV variable is released to the environment by Travis. This isn't done on PR builds.

**Pull request closed without merging**